doc/guides/kconfig/index.rst

```diff
@@ -326,6 +326,108 @@ error-prone, since it can be hard to spot that the same dependency is added
 twice.
 
 
+``depends on`` and ``string``/``int``/``hex`` symbols
+*****************************************************
+
+``depends on`` works not just for ``bool`` symbols, but also for ``string``,
+``int``, and ``hex`` symbols (and for choices).
+
+The Kconfig definitions below will hide the ``FOO_DEVICE_FREQUENCY`` symbol and
+disable any configuration output for it when ``FOO_DEVICE`` is disabled.
+
+.. code-block:: none
+
+   config FOO_DEVICE
+   	bool "Foo device"
+
+   config FOO_DEVICE_FREQUENCY
+   	int "Foo device frequency"
+   	depends on FOO_DEVICE
+
+In general, it's a good idea to check that only relevant symbols are ever shown
+in the ``menuconfig`` interface. Having ``FOO_DEVICE_FREQUENCY`` show up when
+``FOO_DEVICE`` is disabled (and possibly hidden) makes the relationship between
+the symbols harder to understand, even if code never looks at
+``FOO_DEVICE_FREQUENCY`` when ``FOO_DEVICE`` is disabled.
+
+
+``menuconfig`` symbols
+**********************
+
+Defining a symbol with the ``menuconfig`` keyword instead of the ``config``
+keyboard changes how the symbol and its children are displayed in the
+``menuconfig`` interface, putting the children in a separate menu instead of
+indenting them. The children of a symbol are all symbols that immediately
```

Comment:
Should it say "The children of a +menu+ symbol" ?

Reply:
Goes for all symbols, just that they're shown indented instead of in a separate menu when the symbol isn't defined with `menuconfig`.

Was trying to clarify what "children of a symbol" means. Wonder if it could be rephrased.

Reply:
Rephrased it to explain what a child is first. Clearer?

Reply:
Yes, I think the previous text was only unclear to me due to my ignorance